### PR TITLE
A new rule for automatic reconciliation and a cron

### PR DESCRIPTION
**Pull request by nbessi**
This allows for credit lines to use another bank statement as reference, and do a reconciliation with all its lines (debit AND credit, as the commission line can be created in that statement as well).

The main goal is to allow to create "bank lines" for when the payment for credit cards arrives, and reconcile it with a previous bank statement import of those cards.

### CODE REVIEW

File: account_advanced_reconcile_bank_statement/advanced_reconciliation.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            'partner_id',
+            'account_id',
+            'move_id',
+            'statement_id')
+        result = ["account_move_line.%s" % col for col in aml_cols]
+        result += ["account_bank_statement.name as statement_name"]
+        return result
+
+    def _from(self, rec, *args, **kwargs):
+        # Overriden to use a inner join
```
Comment (lepistone) [inline account_advanced_reconcile_bank_statement/advanced_reconciliation.py:50]: Can you elaborate a bit more in the comment on why an inner join?

File: account_advanced_reconcile_bank_statement/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+""",
+ 'version': '1.0.0',
+ 'author': 'Camptocamp',
+ 'category': 'Finance',
+ 'website': 'http://www.camptocamp.com',
+ 'depends': ['account_advanced_reconcile'],
+ 'data': ['easy_reconcile_view.xml'],
+ 'demo': [],
+ 'test': [], # To be ported or migrate to unit tests or scenarios
```
Comment (lepistone) [inline account_advanced_reconcile_bank_statement/__openerp__.py:40]: comment out of place?

**Comment by lepistone** [comment]
Can you provide some automated test, also to show how should this be used in practice? Thanks!

File: account_advanced_reconcile_bank_statement/advanced_reconciliation.py (hunk 1 above)
Comment (nbessi) [inline account_advanced_reconcile_bank_statement/advanced_reconciliation.py:50]: Please ask @mdietrichc2c it is a port of his MP from Launchpad

**Comment by nbessi** [comment]
I'm off for two week, please @mdietrichc2c can you take care of this.

File: account_advanced_reconcile_bank_statement/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ 'auto_install': False,
+ 'installable': True,
+ 'images': []
+}
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lepistone) [inline account_advanced_reconcile_bank_statement/__openerp__.py:45]: I'm not a fan of those editor-specific comments. I do use vim myself but I keep my configuration in my own dotfiles.

A common practice with odoo staff, so I'm not blocking over that.

**Comment by lepistone** [comment]
I rebased this PR as https://github.com/OCA/bank-statement-reconcile/pull/16
Please continue the discussion there.

**Comment by bwrsandman** [comment]
you can `git push -f` after a rebase, you wouldn't need to reopen the PR

**Comment by lepistone** [comment]
@bwrsandman but @nbessi is on holiday and I don't have push access to his repo.

**Comment by bwrsandman** [comment]
Right! Didn't see that detail.

**Pull request closed without merging**